src/Microsoft.AspNet.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -110,27 +105,63 @@ public virtual string RouteUrl(UrlRouteContext routeContext)
             }
 
             var valuesDictionary = new RouteValueDictionary(routeContext.Values);
+            var virtualPathData = GetVirtualPathData(routeContext.RouteName, valuesDictionary);
+            return GenerateUrl(routeContext.Protocol, routeContext.Host, virtualPathData, routeContext.Fragment);
+        }
 
-            var path = GeneratePathFromRoute(routeContext.RouteName, valuesDictionary);
-            if (path == null)
+        /// <inheritdoc />
+        public virtual string Content(string contentPath)
```

Comment:
Hard to see what's changing with this method move in the middle. Could you move it back during the PR and fix it's position when we're done?